src/simulation/ApplyLoad.cpp

```diff
@@ -255,15 +260,21 @@ ApplyLoad::calculateRequiredHotArchiveEntries(Config const& cfg)
     // to scale the expected number of restores by the transaction queue size.
     totalExpectedRestores *= cfg.SOROBAN_TRANSACTION_QUEUE_SIZE_MULTIPLIER;
 
+    if (mode == ApplyLoadMode::FIND_LIMITS_FOR_MODEL_TX)
+    {
+        totalExpectedRestores *= log2(cfg.APPLY_LOAD_MAX_SOROBAN_TX_COUNT);
```

Comment:
A comment here would be helpful explaining that we expect this mode to run a max of log(maxTxCount) # of times, so totalExpectedRestores is increased by that.